miniapps/performance/ex3.cpp

```diff
@@ -66,15 +66,51 @@ typedef ND_FiniteElementSpace<sol_fe_t>       sol_fes_t;
 // Static quadrature, coefficient and integrator types
 typedef TIntegrationRule<geom,ir_order>       int_rule_t;
 typedef TConstantCoefficient<>                coeff_t;
-//typedef TIntegrator<coeff_t,THcurlMassKernel> integ_t;
-typedef TIntegrator<coeff_t,THcurlcurlKernel> integ_t;
+typedef TIntegrator<coeff_t,THcurlMassKernel> mass_integ_t;
+typedef TIntegrator<coeff_t,THcurlcurlKernel> curl_integ_t;
 
 typedef NDShapeEvaluator<sol_fe_t,int_rule_t>  sol_Shape_Eval;
 typedef NDFieldEvaluator<sol_fes_t,ScalarLayout,int_rule_t> sol_Field_Eval;
 
 // Static bilinear form type, combining the above types
-typedef TBilinearForm<mesh_t,sol_fes_t,int_rule_t,integ_t,sol_Shape_Eval,sol_Field_Eval>
-HPCBilinearForm;
+typedef TBilinearForm<mesh_t,sol_fes_t,int_rule_t,
+        mass_integ_t,sol_Shape_Eval,sol_Field_Eval>
+        HPCMassBilinearForm;
+typedef TBilinearForm<mesh_t,sol_fes_t,int_rule_t,
+        curl_integ_t,sol_Shape_Eval,sol_Field_Eval>
+        HPCCurlBilinearForm;
+
+class SumOperator : public Operator
+{
+public:
+   SumOperator(const Operator& a_mass_oper,
+               const Operator& a_der_oper);
+
+   virtual void Mult(const Vector &x, Vector &y) const;
+
+private:
+   const Operator& a_mass_oper_;
+   const Operator& a_der_oper_;
+
+   mutable Vector temp_;
+};
+
+SumOperator::SumOperator(const Operator& a_mass_oper,
+                         const Operator& a_der_oper)
+   :
+   Operator(a_mass_oper.Height()),
+   a_mass_oper_(a_mass_oper),
+   a_der_oper_(a_der_oper),
+   temp_(a_mass_oper.Height())
+{
+}
+
+void SumOperator::Mult(const Vector &x, Vector &y) const
+{
+   a_mass_oper_.Mult(x, y);
+   a_der_oper_.Mult(x, temp_);
```

Comment:
It's not really important. Wouldn't it be better to avoid this temporary? I know there isn't AddMult in Operator class, I'm not sure removing this temporary would do a big difference though.